src/ui/imageproviders.cpp

```diff
@@ -87,24 +98,24 @@ QImage getMetatileImage(
             tile_image.setColor(0, color.rgba());
         }
 
-        metatile_painter.drawImage(origin, tile_image.mirrored(tile_.xflip, tile_.yflip));
+        metatile_painter.drawImage(origin, tile_image.mirrored(tile.xflip, tile.yflip));
     }
     metatile_painter.end();
 
     return metatile_image;
 }
 
-QImage getTileImage(uint16_t tile, Tileset *primaryTileset, Tileset *secondaryTileset) {
-    Tileset *tileset = Tileset::getBlockTileset(tile, primaryTileset, secondaryTileset);
-    int local_index = Metatile::getBlockIndex(tile);
+QImage getTileImage(uint16_t tileId, Tileset *primaryTileset, Tileset *secondaryTileset) {
+    Tileset *tileset = Tileset::getBlockTileset(tileId, primaryTileset, secondaryTileset);
+    int local_index = Metatile::getBlockIndex(tileId);
     if (!tileset) {
         return QImage();
     }
     return tileset->tiles.value(local_index, QImage());
 }
 
-QImage getColoredTileImage(uint16_t tile, Tileset *primaryTileset, Tileset *secondaryTileset, QList<QRgb> palette) {
-    QImage tileImage = getTileImage(tile, primaryTileset, secondaryTileset);
+QImage getColoredTileImage(uint16_t tileId, Tileset *primaryTileset, Tileset *secondaryTileset, QList<QRgb> palette) {
+    QImage tileImage = getTileImage(tileId, primaryTileset, secondaryTileset);
     if (tileImage.isNull()) {
         tileImage = QImage(16, 16, QImage::Format_RGBA8888);
```

Comment:
The fallback dimensions in this function are mistakenly (I believe) using the dimensions of an entire metatile rather than of a single tile. This can cause some unexpected behavior with `addTileImage`. If there wasn't a reason for this I'll change them to `8`.